packages/react/src/animation/TextRevealAnimation/TextRevealAnimation.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {useCallback, useEffect} from 'react'
+import clsx from 'clsx'
+import {useTextRevealAnimation} from './useTextRevealAnimation'
+
+import styles from './TextRevealAnimation.module.css'
+
+export function TextRevealAnimation({children, ...rest}) {
+  const [animationStarted, setAnimationStarted] = React.useState(false)
```

Comment:
I'm surprised that TS isn't complaining here, but we're missing a type on the props

```suggestion
type TextRevealAnimationProps = React.HTMLAttributes<HTMLSpanElement> & {children: string}
```

We'll either have to force `children` to be a string here, or accept the normal children and have the hook check that it's a string and nothing else

Reply:
Thanks. I've added a type declaration for TS users and a warning for JS., which will cover most use-cases. We don't use inline types, so exporting a dedicated declaration for it instead.